libs/langchain-community/src/embeddings/yandex.ts

```diff
@@ -0,0 +1,151 @@
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+import { Embeddings, EmbeddingsParams } from "@langchain/core/embeddings";
+
+const apiUrl =
+  "https://llm.api.cloud.yandex.net/foundationModels/v1/textEmbedding";
+
+export interface YandexGPTEmbeddingsParams extends EmbeddingsParams {
+  /** Model name to use. */
+  model?: string;
+
+  /** Model version to use. */
+  modelVersion?: string;
+
+  /** Model version to use. */
+
+  /** Model URI to use. */
+  modelURI?: string;
+
+  /** Yandex Cloud Folder ID. */
+  folderID?: string;
+
+  /**
+   * Yandex Cloud Api Key for service account
+   * with the `ai.languageModels.user` role.
+   */
+  apiKey?: string;
+
+  /**
+   * Yandex Cloud IAM token for service or user account
+   * with the `ai.languageModels.user` role.
+   */
+  iamToken?: string;
+}
+
+/**
+ * Class for generating embeddings using the YandexGPT Foundation models API. Extends the
+ * Embeddings class and implements GradientEmbeddingsParams and
+ */
+export class YandexGPTEmbeddings
+  extends Embeddings
+  implements YandexGPTEmbeddingsParams
+{
+  model = "text-search-query";
+
+  modelVersion = "latest";
+
+  modelURI?: string;
+
+  apiKey?: string;
+
+  iamToken?: string;
+
+  folderID?: string;
+
+  constructor(fields?: YandexGPTEmbeddingsParams) {
+    super(fields ?? {});
+
+    const apiKey = fields?.apiKey ?? getEnvironmentVariable("YC_API_KEY");
+
+    const iamToken = fields?.iamToken ?? getEnvironmentVariable("YC_IAM_TOKEN");
+
+    const folderID = fields?.folderID ?? getEnvironmentVariable("YC_FOLDER_ID");
+
+    if (apiKey === undefined && iamToken === undefined) {
+      throw new Error(
+        "Please set the YC_API_KEY or YC_IAM_TOKEN environment variable or pass it to the constructor as the apiKey or iamToken field."
+      );
+    }
+
+    this.modelURI = fields?.modelURI;
+    this.apiKey = apiKey;
+    this.iamToken = iamToken;
+    this.folderID = folderID;
+    this.model = fields?.model ?? this.model;
+    this.modelVersion = fields?.modelVersion ?? this.modelVersion;
+
+    if (this.modelURI === undefined && folderID === undefined) {
+      throw new Error(
+        "Please set the YC_FOLDER_ID environment variable or pass Yandex GPT model URI to the constructor as the modelURI field."
+      );
+    }
+
+    if (!this.modelURI) {
+      this.modelURI = `emb://${this.folderID}/${this.model}/${this.modelVersion}`;
+    }
+  }
+
+  /**
+   * Method to generate embeddings for an array of documents.
+   * @param texts Array of documents to generate embeddings for.
+   * @returns Promise that resolves to a 2D array of embeddings for each document.
+   */
+  async embedDocuments(texts: string[]): Promise<number[][]> {
+    return this.caller.call(async () => {
```

Comment:
Small nit, in other embeddings classes we wrap the calls with an `embeddingWithRetry` method. Not exactly necessary in this case, but could prove helpful in the future.
[Example](https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-mistralai/src/embeddings.ts#L133C3-L133C36)